docs/graphcast/radios/intro.md

```diff
@@ -31,251 +28,196 @@ Once that's done, you can start building your very first Radio.
-    const message = await gossipAgent.processMessage({
-      msg,
-      topic,
-      types: RADIO_PAYLOAD_TYPES,
-    });
-
-    logger.info(`Message: `, { message });
-    const radioPayload = JSON.parse(message.radioPayload);
-
-    if (radioPayload.content === "Ping") {
-      const radioPayload = {
-        content: "Pong",
-      };
-
-      await sendMessage(radioPayload);
+We'll define a helper function that holds the logic of sending messages to the Graphcast network:
+
+```Rust
+async fn send_message(content: String, block_number: u64) {
+        match GOSSIP_AGENT
+            .get()
+            .unwrap()
+            .send_message(
+                "ping-pong-content-topic".to_string(),
+                block_number,
+                content,
+            )
+            .await
+        {
+            Ok(sent) => println!("Sent message id:: {}", sent),
+            Err(e) => println!("Failed to send message: {}", e),
+        };
     }
-  } catch (error) {
-    logger.info(`Failed to handle the message, moving on. ${error}`);
-  }
-};
-
-await gossipAgent.establishTopics(DOMAIN, handler);
 ```
 
-The `handler` function accepts a message of type `Uint8Array` (bytes), and of course the topic it was received on. At this point, the message is still encoded, but we can decode it using the SDK helper method `processMessage` with the `RADIO_PAYLOAD_TYPES` variable we defined before (because it was also used to encode it on the sender side). After it's decoded we can inspect the incoming `radioPayload` and read the `content` variable. Right after the function definition we can pass it to `establishTopics`. We've now started listening to messages 👂.
+Again, the `identifier` that we define as `ping-pong-content-topic` can be any string that suits your Radio logic, if it doesn't really matter for your use case (like in the ping-pong Radio case) you can just use a UUID or a hardcoded string.
 
-### The main loop
+### Receiving and handling messages
 
-Great, we're almost there! We have a way to pass messages back and forth 🏓. But sending a one-off message is no fun, we want to create some sort of scheduled and continuous logic of message exchange, and perhaps the easiest way to do that is to use the Ethereum block number as cue. For instance we can set a schedule where the message is sent when we hit a block that is divisible by 5.
+We now know how to send message, but how do we receive and handle message from other network participants?
 
-```typescript
-gossipAgent.clientManager.ethClient.provider.on("block", async (block) => {
-  logger.info(`🔗 Block: ${block}`);
+We need to define a custom callback handler that specifies what to do with incoming messages. There cannot be any async code inside the handler, that's why we're saving the incoming message (after it has been validated by the SDK) for later processing, where we will check it's content and perform some action based on it.
```

Comment:
Maybe we can emphasize on the flexibility of how they store the messages? 
`After gossip agent validate the incoming messages, we provide a custom callback handler that specifies what to do with the message. In this handler we cache the message for later aggregation and attestation, but depending you are free to use a database or self defined data structures.`